[FIX] tools: fix orientation of image with EXIF orientation tag

seb-odoo opened the pull request:
Before this commit, some images would display incorrectly orientated.

This typically happens for images taken from a non-standard orientation
by some phones or other devices that are able to report orientation.

The specified transposition is applied to the image before all other
operations, because all of them expect the image to be in its final
orientation, which is the case only when the first row of pixels is the top
of the image and the first column of pixels is the left of the image.

Moreover the EXIF tags will not be kept when the image is later saved, so
the transposition has to be done to ensure the final image is correctly
orientated.

seb-odoo commented:
Note that since https://github.com/python-pillow/Pillow/pull/4063 Pillow seems to do it by default, but only for TIFF files. Might need to add an ugly hack to avoid doing the job when Pillow already does it if we want to support multiple versions at the same time.

seb-odoo commented:
@mart-e I would like your opinion on this PR and also the above comment.

mart-e commented:
> Might need to add an ugly hack to avoid doing the job when Pillow already does it if we want to support multiple versions at the same time.

We do that [from time to time](https://github.com/odoo/odoo/blob/8ec307d81cf0f2f62f5b27668d737097aeb25350/odoo/sql_db.py#L47), it is acceptable.

(still need to review the patch)

seb-odoo commented:
TODO: 

`            if hasattr(ImageOps, 'exif_transpose'):
                # added in pillow 6.0
                return ImageOps.exif_transpose(image)`

maybe also add a condition on type jpeg for the condition on getexif

seb-odoo commented:
self todo also retarget to 13 since 12.4 is dead

seb-odoo commented:
I had to make some adaptations: mainly extracting `image_fix_orientation` out of `base64_to_image` because it would lose the original format of the image in the process.

Hopefully this should be good now, rebased, re-targeted, and added the Pillow 6 function.

I don't really like adding a link in the commit message as it could always become broken at some point. I think I have commented enough, and this PR will be linked anyway.

seb-odoo commented:
@robodoo r+

robodoo commented:
Staging failed: ci/runbot on 26f631a6ad9b334f1a8f061c6d45a1192ac377db (view more at http://runbot.odoo.com/runbot/build/651601)

mart-e commented:
robodoo retry

robodoo commented:
Merged at 0f8e132ec0495fcdfa0a47d5b9c98a2f6f10c7d8, thanks!
